module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -1026,10 +1029,10 @@ public function editAction()
         // to a particular list ID:
         $containingLists = [];
         if (!empty($listID)) {
-            $userResources = $user->getSavedData($id, null, $source);
+            $userResources = $userResourceService->getFavoritesForRecord($id, $source, null, $user);
         }
         foreach ($userResources as $current) {
-            $containingLists[] = $current->list_id;
+            $containingLists[] = $current->getUserList()->getId();
```

Comment:
`null` is a legal return value for `getUserList()`, do we need to take that into account here?